src/pyOpenMS/pxds/IsotopeLabelingMDVs.pxd

```diff
@@ -25,7 +26,7 @@ cdef extern from "<OpenMS/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.h>" namespac
 
         void calculateIsotopicPurities(
           const FeatureMap & normalized_feature,
-          const libcpp_vector[ libcpp_vector[double] ] & experiment_data, const libcpp_vector[String] & isotopic_purity_name) nogil except +
+          const libcpp_vector[ DoubleList ] & experiment_data, const libcpp_vector[String] & isotopic_purity_name) nogil except +
```

Comment:
hmm maybe this doesn't work because https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/pxds/DoubleList.pxd
it is a simple typedef?
What is the correct way? 
Introducing a class that publicly derives from vector<double>? e.g., class GoodNameForTheDoubleList : public DoubleList { }; ? 


```
Traceback (most recent call last):
  File "create_cpp_extension.py", line 200, in <module>
    autowrap_include_dirs = doCythonCodeGeneration(modname, allDecl_mapping, instance_map, converters)
  File "create_cpp_extension.py", line 164, in doCythonCodeGeneration
    include_boost=False, include_numpy=True, allDecl=allDecl_mapping)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/__init__.py", line 65, in generate_code
    gen.create_pyx_file(debug)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 215, in create_pyx_file
    create_for(ResolvedClass, self.create_wrapper_for_class)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 211, in create_for
    method(resolved)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 470, in create_wrapper_for_class
    codes = self.create_wrapper_for_method(r_class, name, methods)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 621, in create_wrapper_for_method
    code = self.create_wrapper_for_nonoverloaded_method(cdcl, py_name, methods[0])
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 784, in create_wrapper_for_nonoverloaded_method
    method)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 664, in _create_fun_decl_and_input_conversion
    conv_code, call_as, cleanup = converter.input_conversion(t, n, arg_num)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/ConversionProvider.py", line 670, in input_conversion
    raise Exception("Converter can not handle wrapped classes as keys and values in map")
Exception: Converter can not handle wrapped classes as keys and values in map
src/pyOpenMS/CMakeFiles/pyopenms_create_cpp.dir/build.make:57: recipe for target 'src/pyOpenMS/CMakeFiles/pyopenms_create_cpp' failed
make[3]: *** [src/pyOpenMS/CMakeFiles/pyopenms_create_cpp] Error 1
```

Reply:
@jpfeuffer or @hroest

Reply:
Hmmm mit sure if anything is possible here. I need to check but since cython tries to put it in a dict, you probably have to do manual python code to specify how to hash etc.

Reply:
how do you mean? adding an extra class on the c++ side that wraps the vector<double> should work I think.
Or changing our definition of DoubleList to be a class not a typedef ?

Reply:
I think the problem is in calculateMDVAccuracies with the map of OpenMS::String to OpenMS::String.

Reply:
@jpfeuffer it was commented but that has yielded the same build error : https://github.com/OpenMS/OpenMS/pull/5122/commits/f5def7def6b2d57cbbf3456c8693d916d5cc3fd6

Reply:
But the commit that you linked passed? Or am I seeing something wrong?

Reply:
@jpfeuffer for some odd reason this has been overseen, are there any known limitations to `libcpp_map` ?

Reply:
Yes it somehow has to have a basic type as either key or value. Unfortunately, I don't know exactly why. You could use std::string in that method and cast/convert from String or you create a wrapper class e.g. StringMap.

Reply:
@jpfeuffer in the last [commit](https://github.com/OpenMS/OpenMS/pull/5122/commits/971485216f62d8c2cb7d577062ad561055d37ad7) `calculateMDVAccuracies ` is commented out but the build for pyOMS still fails. `libcpp_string` and `String` has both been tried out and with no success.